sdk/core/Azure.Core.Experimental/samples/Variant.md

```diff
@@ -0,0 +1,135 @@
+- `float`
+- `float?`
+- `double`
+- `double?`
+- `DateTimeOffset`
+- `DateTimeOffset?`
+- `DateTime`
+- `DateTime?`
+- Enums
+
+</details>
+
+In principle, `Variant` is similar to `object`, but without boxing as described above.  However, since it's not `object`, there are different APIs needed to achieve some of the same functionality, including:
+
+- [Assigning a value to `Variant`](#assign-a-value-to-variant)
+- [Retrieving the value a `Variant` holds](#get-the-value-from-variant)
+- [Working with `null` and `Variant`](#handling-nulls)
+
+## Assign a value to Variant
+
+`Variant` has implicit cast operators for each of the primitives listed above.  This means you can assign a value of those types to `Variant` without casting, and a new `Variant` instance holding an unboxed copy of the value will be created.
+
+```csharp
+Variant v = 3L;
+
+// v.Type is System.Int64
+```
+
+Since you can't have an implicit cast operator from `object`, for reference types, you must create a new instance of `Variant` and pass the value to the `Variant` constructor.
+
+```csharp
+MemoryStream s = new();
+Variant v = new(s);
+
+// v.Type is System.IO.MemoryStream
+```
+
+`Variant` will box value types that aren't on the list above, because it only stores 16 bytes and user-defined value types might exceed this size.  The one exception is for `enum`, which `Variant` can hold without boxing.  Because we can't add implicit cast operators for unknown enum types, you must call the `Create` method to create a `Variant` that holds an enum without boxing.
+
+```csharp
+Value v = Value.Create(System.Color.Blue);
+
+// v.Type is System.Color
+```
+
+## Get the value from Variant
+
+`Variant` has explicit cast operators for each of the primitives listed above, as well as to `string`.  That means you can assign a `Variant` holding one of these types to a variable of that type with a cast:
+
+```csharp
+Variant v = 3;
+int i = (int)v;
+```
+
+If you don't know the type of the value that a `Variant` is holding, you can read that from the `Variant.Type` property.
+
+```csharp
+switch (v.Type)
+{
+    case Type s when s == typeof(string):
+        Console.WriteLine($"string: {v}");
+        break;
+    case Type i when i == typeof(int):
+        Console.WriteLine($"int: {(int)v}");
+        break;
+}
+```
+
+If you do know the type the `Variant` is holding, you can use the `As<T>` method to get its value as that type.
+
+```csharp
+MemoryStream s = new();
+Variant v = new (s);
+
+MemoryStream streamValue = v.As<MemoryStream>();
+```
+
+If you call `Variant.As<T>` and the `Variant` isn't holding the type that you ask for, it will throw an `InvalidCastException`.  To try to retrieve the value without risking an exception being thrown, you can use the `TryGetValue` method instead.
+
+```csharp
+if (v.TryGetValue(out string s))
+{
+    Console.WriteLine($"string: {s}");
+}
+
+if (v.TryGetValue(out int i))
+{
+    Console.WriteLine($"int: {i}");
+}
+```
+
+## Handling nulls
```

Comment:
Avoid gerunds in headings.
```suggestion
## Handle nulls
```

Reply:
What's the thinking being this rule?

Reply:
I won't die on this hill, but there are 3 reasons:
1. It's more difficult for non-English speakers to parse gerunds, since many of their languages don't have this word form. This factor matters if/when the content is localized. For example, if we decide to publish this content to learn.microsoft.com down the road.
2. Gerunds are longer.
3. The indicative verb form is the most direct (create a database vs. creating a database).